src/runtime/particle-execution-host.ts

```diff
@@ -57,7 +57,10 @@ export class ParticleExecutionHost {
 
   constructor({slotComposer, arc, ports}: ParticleExecutionHostOptions) {
     this.close = () => {
-      this._apiPorts.forEach(apiPort => apiPort.close());
+      this._apiPorts.forEach(apiPort => {
```

Comment:
Suggest we make a stub for `clear()` in APIPort, and then call `clear()` inside of `close()`. Then we can avoid the typecast and simplify the calling code here in ParticleExecutionHost.

Reply:
Good suggestion, thank you.
Added a stub in APIPort and remove all type casts!

Reply:
Also called clear inside close.

Reply:
Done!